runners/core-java/src/test/java/org/apache/beam/runners/core/ReduceFnRunnerTest.java

```diff
@@ -140,7 +141,40 @@ public Void answer(InvocationOnMock invocation) throws Exception {
       }
     })
     .when(mockTrigger).onFire(anyTriggerContext());
- }
+  }
+
+  /**
+   * Tests that a processing time timers does not cause window GC.
+   */
+  @Test
+  public void testProcessingTimeTimersDoesNotGc() throws Exception {
```

Comment:
@bjchambers some grammar errors... but this one fires  a processing time timer with a timestamp long after the end of the window, but no watermark advancement. This test does fail without the change to `ReduceFnRunner`, confirming that it definitely causes a GC.